src/aws.ts

```diff
@@ -1,3 +1,32 @@
+export const awsProfileGenerator: Fig.Generator = {
+  cache: {
+    strategy: "stale-while-revalidate",
+    cacheByDirectory: true,
+  },
+  script: "aws configure list-profiles",
+  postProcess: function (out, [awsClient]) {
+    if (out.trim() == "") {
+      return [];
+    }
+
+    try {
+      const lines = out.split("\n");
+
+      if (lines) {
+        return lines.map((line) => {
+          return {
+            name: line,
+          };
+        });
+      }
+    } catch (e) {
+      console.error(e);
+    }
+
+    return [];
```

Comment:
`trycatch` does nothing in this case and `String.prototype.split` always returns an array of at least one value if the string exists:
```suggestion
    const lines = out.split("\n");
    return lines.map((line) => {
      return {
        name: line,
      };
    });
```